aces/Loris
# Create script to validate server configuration

## Brief summary of changes

@christinerogers recently presented some features of REDCap. Many of them seem beneficial to LORIS especially regarding set-up and installation.

One feature she recommended was creating a "Configuration Check". In REDCap, this does things like verifying minimum versions of dependencies, validating that directories exists and are writable, testing email, and so on.

This is an effort to implement this in LORIS using a PHP script in the `tools/` directory.

In the future I think this could be expanded into a module for front-end Administrators to check that their LORIS version is working properly.

Output:
<img width="961" alt="Screen Shot 2020-02-07 at 14 29 59" src="https://user-images.githubusercontent.com/4022790/74059608-5ceaf900-49b6-11ea-9813-da6b5ee836f1.png">

#### Things to check:

##### Minimum versions of our requirements
- [x] PHP
- [x] Apache 
- [x] Composer 
- [x] NPM
- [x] MySQL/MariaDB

##### Web Paths
- [x] Do the directories exist and are they writable?

##### Connections to web-based dependencies
- ~~Google reCAPTCHA~~ This isn't working in LORIS at the moment but should be added in the future.
- [x] "Have I Been Pwned?", etc.

##### Outgoing email
- [x] Sends a test email

For now I am creating a script in `tools` to validate some settings in LORIS. Eventually I think this would be a nice module to include under `Admin`. But a tools script is a good start.

I'm also adding nice coloured output like I have in #5260 as well as logging via a "Helper class". This isn't necessary but I think it helps a lot and can be used by other tools.

Future work includes adding checks for PHP extensions.

### Testing instructions (if applicable)

Checkout the branch.
`cd` to the `tools/` directory and run the script. Make sure the output makes sense.

### Related issues

Resolves #5021 
Also resolves https://github.com/aces/Loris/issues/6811
